### ISSUE TITLE
Delivery slip locks order, invoice, payment, status does not

### ISSUE THREAD

**Issue by madsoliver** [PrestaShop/PrestaShop#19043]
Orders become un-editable “generate delivery slip”.
Invoice, payment, status, "consider valid", or anything else affects if an order is editable.

Why is this a problem? When a merchant wants to pick and ship an order, he needs a delivery slip. When generating this the order gets locked. When then the merchant discovers that one of the items in the order is out of stock, he wants to change the order - but cannot.

I understand that an order needs to be locked after generating a legal document like the invoice, but when it comes to the invoice, this actually does not lock the order.

**I don't get the logic.**

As discussed with @MatShir on Slack, there should be made a workflow diagram for order management. 

The way it is now, it’s somewhere in between do whatever you want, make whatever statuses you want and you can’t do this and that, because of this and that. Maybe in AdminStatuses  there should be more checkboxes where you can choose more freely what the statuses should do?
Examples: “Make order un-editable with and after this status”, “Make credit slip for the entire order on this status”, “Disallow status change after this status”.
Also the wording on this page is unclear (and lacks documentation). Consider the associated order as validated. <- What is it, what does it do? Show delivery PDF. <- Actually it does alot more than just show the PDF, it locks the order :wink:

**Comment by atomiix** [comment]
Hi @onklmaps 

Thanks for your feedback, I'll let the @PrestaShop/prestashop-product-team know about it.

**Comment by colinegin** [comment]
@onklmaps @atomiix can you confirm this behaviour is already present in 1.7.6 ?

**Comment by atomiix** [comment]
@colinegin yes this behaviour is present in 1.7.6 because in order to get the delivery slip, your order status should be "shipped".

**Comment by madsoliver** [comment]
Any news on this? 

Is it fixed in 1.7.7.x?

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team friendly reminder :wink:

**Comment by madsoliver** [comment]
I made a discussion for this: 
https://github.com/PrestaShop/PrestaShop/discussions/31038
